File: libs/toolkit-android/src/test/java/com/example/android/utils/LogUtilityTest.java
package com.example.android.utils;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;
import org.robolectric.annotation.Config;

@RunWith(RobolectricTestRunner.class)
@Config(manifest = Config.NONE)
public class LogUtilityTest {
  @Test
  public void shouldNotFailWhenAnyLogMethodIsCalled() {
    LogUtility.d(new NullPointerException("Sample Null Pointer Exception"));
    LogUtility.d("Log", "Arg 1", "Arg 2");
    LogUtility.d(new NullPointerException("Sample Null Pointer Exception"), "Log", "Arg 1", "Arg 2");

    LogUtility.e(new NullPointerException("Sample Null Pointer Exception"));
    LogUtility.e("Log", "Arg 1", "Arg 2");
    LogUtility.e(new NullPointerException("Sample Null Pointer Exception"), "Log", "Arg 1", "Arg 2");

    LogUtility.w(new NullPointerException("Sample Null Pointer Exception"));
    LogUtility.w("Log", "Arg 1", "Arg 2");
    LogUtility.w(new NullPointerException("Sample Null Pointer Exception"), "Log", "Arg 1", "Arg 2");
  }
}

File: android-utils/src/test/java/com/example/android/utils/LogUtilityTest.kt
package com.example.android.utils

import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner
import org.robolectric.annotation.Config

@RunWith(RobolectricTestRunner::class)
@Config(manifest = Config.NONE)
class LogUtilityTest {
	@Test
	fun shouldNotFailWhenAnyLogMethodIsCalled() {
		LogUtility.d(NullPointerException("Sample Null Pointer Exception"))
		LogUtility.d("Log", "Arg 1", "Arg 2")
		LogUtility.d(NullPointerException("Sample Null Pointer Exception"), "Log", "Arg 1", "Arg 2")

		LogUtility.e(NullPointerException("Sample Null Pointer Exception"))
		LogUtility.e("Log", "Arg 1", "Arg 2")
		LogUtility.e(NullPointerException("Sample Null Pointer Exception"), "Log", "Arg 1", "Arg 2")

		LogUtility.w(NullPointerException("Sample Null Pointer Exception"))
		LogUtility.w("Log", "Arg 1", "Arg 2")
		LogUtility.w(NullPointerException("Sample Null Pointer Exception"), "Log", "Arg 1", "Arg 2")
	}
}
